app/Mail/Services/UnsubscribeService.php

```diff
@@ -142,7 +142,15 @@ public function processUnsubscribe(string $token, bool $shouldGenerateUndoToken
             $descriptionData = $this->getGranularDescription($data->subjectType, $data->subjectId);
         } elseif ($data instanceof CategoryUnsubscribeData) {
             $currentPrefs = $user->websitePrefs;
-            $newPrefs = $currentPrefs & ~(1 << $data->preference);
+
+            // For "*Off_*" preferences, unsubscribing means setting the bit.
+            // For "*On_*" preferences, unsubscribing means clearing the bit.
+            if ($data->preference === UserPreference::EmailOff_DailyDigest) {
+                $newPrefs = $currentPrefs | (1 << $data->preference);
+            } else {
+                $newPrefs = $currentPrefs & ~(1 << $data->preference);
+            }
```

Comment:
This might make more sense as a switch statement:
```
switch ($data->preference) {
    case UserPreference::EmailOff_DailyDigest:
        // For "*Off_*" preferences, unsubscribing means setting the bit.
        $newPrefs = $currentPrefs | (1 << $data->preference);
        break;
    default:
        // For "*On_*" preferences, unsubscribing means clearing the bit.
        $newPrefs = $currentPrefs & ~(1 << $data->preference);
        break;
}
```